fabric-network-v1/src/main/java/net/fabricmc/fabric/api/network/v1/receiver/PacketContext.java

```diff
@@ -0,0 +1,97 @@
+/*
+ * Copyright (c) 2016, 2017, 2018, 2019 FabricMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.fabricmc.fabric.api.network.v1.receiver;
+
+import net.minecraft.client.MinecraftClient;
+import net.minecraft.client.network.ClientLoginNetworkHandler;
+import net.minecraft.client.network.ClientPlayNetworkHandler;
+import net.minecraft.network.listener.PacketListener;
+import net.minecraft.server.MinecraftServer;
+import net.minecraft.server.network.ServerLoginNetworkHandler;
+import net.minecraft.server.network.ServerPlayNetworkHandler;
+import net.minecraft.util.PacketByteBuf;
+import net.minecraft.util.thread.ThreadExecutor;
+
+import net.fabricmc.fabric.api.network.v1.sender.LoginQueryPacketSender;
+import net.fabricmc.fabric.api.network.v1.sender.CustomPayloadPacketSender;
+
+/**
+ * Interface defining a context used during packet processing. Allows access
+ * to additional information, such as the source/target of the player, or
+ * the correct task queue to enqueue synchronization-requiring code on.
+ */
+public interface PacketContext {
+	/**
+	 * Get the task queue for a given side.
+	 *
+	 * <p>As Minecraft networking I/O is asynchronous, but a lot of its logic is
+	 * not thread-safe, it is recommended to do the following:
+	 *
+	 * <ul><li>read and parse the PacketByteBuf,
+	 * <li>run the packet response logic through the main thread task queue via
+	 * ThreadTaskQueue.execute(). The method will check if it's not already
+	 * on the main thread in order to avoid unnecessary delays, so don't
+	 * worry about that!</ul>
+	 *
+	 * @return The thread task queue.
+	 */
+	ThreadExecutor<?> getTaskQueue();
```

Comment:
I assume there is no extra functionality you really need to expose from a NonBlockingThreadExecutor based on the fact you've only gone up to the the ThreadExecutor.

Reply:
hmm, i am exposing the server/client for the login users